src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -1152,5 +1210,23 @@ IReadOnlyList<Type> IInvocationBinder.GetParameterTypes(string methodName)
                 throw new InvalidOperationException($"There are no callbacks registered for the method '{methodName}'");
             }
         }
+
+        private class PeriodicConnectionTimer
```

Comment:
I think this class may be a little overkill 😄 . The only things we need to store are the interval (which we already have on HubConnection) and the next activation (which we could just put straight on HubConnection), and the `Reset` method is pretty simple, so we can just inline it wherever it's used.